Emby.Server.Implementations/Library/Validators/CollectionPostScanTask.cs

```diff
@@ -0,0 +1,150 @@
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using MediaBrowser.Controller.Collections;
+using MediaBrowser.Controller.Entities;
+using MediaBrowser.Controller.Entities.Movies;
+using MediaBrowser.Controller.Library;
+using MediaBrowser.Model.Querying;
+using Jellyfin.Data.Enums;
+using Microsoft.Extensions.Logging;
+using MediaBrowser.Model.Entities;
+
+namespace Emby.Server.Implementations.Library.Validators
+{
+    /// <summary>
+    /// Class CollectionPostScanTask.
+    /// </summary>
+    public class CollectionPostScanTask : ILibraryPostScanTask
+    {
+        private readonly ILibraryManager _libraryManager;
+        private readonly ICollectionManager _collectionManager;
+        private readonly ILogger<CollectionPostScanTask> _logger;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CollectionPostScanTask" /> class.
+        /// </summary>
+        /// <param name="libraryManager">The library manager.</param>
+        /// <param name="collectionManager">The collection manager.</param>
+        /// <param name="logger">The logger.</param>
+        public CollectionPostScanTask(
+            ILibraryManager libraryManager,
+            ICollectionManager collectionManager,
+            ILogger<CollectionPostScanTask> logger)
+        {
+            _libraryManager = libraryManager;
+            _collectionManager = collectionManager;
+            _logger = logger;
+        }
+
+        /// <summary>
+        /// Runs the specified progress.
+        /// </summary>
+        /// <param name="progress">The progress.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>Task.</returns>
+        public async Task Run(IProgress<double> progress, CancellationToken cancellationToken)
+        {
+            var boxSets = _libraryManager.GetItemList(new InternalItemsQuery
+            {
+                IncludeItemTypes = new[] { nameof(BoxSet) },
+                CollapseBoxSetItems = false,
+                Recursive = true
+            });
+
+            var collectionNameMoviesMap = new Dictionary<string, HashSet<Guid>>();
```

Comment:
Move the query down below to save a query if there's no items
```suggestion
            var collectionNameMoviesMap = new Dictionary<string, HashSet<Guid>>();
```

Reply:
Fixed, thanks.